### PR TITLE
Dockerized the project

### PR DESCRIPTION
**Pull request by alija83**
Added Docker files

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+FROM debian:bookworm-slim
+
+# Install necessary runtime dependencies
+RUN apt-get update && apt-get install -y \
+    libssl-dev \
+    libclang-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy the built binary from the builder stage
+COPY --from=builder /app/target/release/sabledb /usr/local/bin/sabledb
```
Comment (sabledb-io) [inline Dockerfile:30]: I would actually install it under `/opt/sabledb`
With the following directory structure:

```
/opt/sabledb/bin
/opt/sabledb/log
/opt/sabledb/etc
/opt/sabledb/db
```

Copy the following binaries:

```
target/release/sabledb -> /opt/sabledb/bin
target/release/sdb-cli -> /opt/sabledb/bin
target/release/sb ->  /opt/sabledb/bin
```

In addition, generate `sabledb.ini` file under `/opt/sabledb/etc`:

```bash
cat << EOF >> /opt/sabledb/etc/sabledb.ini
logdir = "/opt/sabledb/log"
listen_ip = "0.0.0.0"
db_path = "/opt/sabledb/db"
config_dir = "/opt/sabledb/etc"
EOF
```

And change the entry point to be:

```
["/opt/sabledb/bin/sabledb", "/opt/sabledb/etc/sabledb.ini"]
```

**Review by sabledb-io — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (sabledb-io) [reply Dockerfile:30]: Actually, I am happy with these changes. I am going to merge this one PR and I am going to modify it a bit later by adding a template `server.docker.ini` file in the repository to reduce the amount of work the user needs to do later on

**Review by sabledb-io — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (alija83) [reply Dockerfile:30]: Sure, sounds good.

**Comment by sabledb-io** [comment]
I merged it and reduced the amount of work user needs to do in order to use this. Thanks for the PR!

**Pull request merged**